modules/docs/arrow-docs/docs/docs/typeclasses/foldable/README.md

```diff
@@ -9,8 +9,520 @@ permalink: /docs/typeclasses/foldable/
+```
+
+### ReduceLeftToOption
+
+```kotlin:ank:silent
+fun <F> reduceLeftToOption(strKind: Kind<F, String>, FO: Foldable<F>): Option<Int> =
+  FO.run {
+    strKind.reduceLeftToOption({ it.length }) { base: Int, value: String -> base + value.length }
+  }
+```
+
+```kotlin:ank
+reduceLeftToOption(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+reduceLeftToOption(None, Option.foldable())
+```
+
+```kotlin:ank
+reduceLeftToOption(strList, ListK.foldable())
+```
+
+### ReduceRightToOption
+
+```kotlin:ank:silent
+fun <F> reduceRightToOption(strKind: Kind<F, String>, FO: Foldable<F>): Option<Int> =
+  FO.run {
+    strKind.reduceRightToOption({ it.length }) { value: String, base: Eval<Int> -> base.map { it + value.length } }
+      .value()
+  }
+```
+
+```kotlin:ank
+reduceRightToOption(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+reduceRightToOption(None, Option.foldable())
+```
+
+```kotlin:ank
+reduceRightToOption(strList, ListK.foldable())
+```
+
+### ReduceLeftOption
+Reduce the elements of this structure down to a single value by applying the provided aggregation function in
+a left-associative manner.
+
+Return None if the structure is empty, otherwise the result of combining the cumulative left-associative result
+of the f operation over all of the elements.
+
+```kotlin:ank:silent
+fun <F> getLengthFromLeft(strKind: Kind<F, String>, FO: Foldable<F>): Option<String> =
+  FO.run {
+    strKind.reduceLeftOption { base: String, value: String -> base + value }
+  }
+```
+
+```kotlin:ank
+getLengthFromLeft(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromLeft(None, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromLeft(strList, ListK.foldable())
+```
+
+### ReduceRightOption
+Reduce the elements of this structure down to a single value by applying the provided aggregation function in
+a right-associative manner.
+
+Return None if the structure is empty, otherwise the result of combining the cumulative right-associative
+result of the f operation over the A elements.
+
+```kotlin:ank:silent
+fun <F> getLengthFromRight(strKind: Kind<F, String>, FO: Foldable<F>): Option<String> =
+  FO.run {
+    strKind.reduceRightOption { value: String, base: Eval<String> -> base.map { it + value } }
+      .value()
+  }
+```
+
+```kotlin:ank
+getLengthFromRight(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromRight(None, Option.foldable())
+```
+
+```kotlin:ank
+getLengthFromRight(strList, ListK.foldable())
+```
+
+### FoldMap
+Fold implemented by mapping A values into B and then combining them using the given `Monoid<B>` instance.
+
+```kotlin:ank:silent
+fun <F> getLenght(strKind: Kind<F, String>, FO: Foldable<F>): Int =
+  FO.run {
+    strKind.foldMap(Int.monoid()) { it.length }
+  }
+```
+
+```kotlin:ank
+getLenght(maybeStr, Option.foldable())
+```
+
+```kotlin:ank
+getLenght(None, Option.foldable())
+```
+
+```kotlin:ank
+getLenght(strList, ListK.foldable())
+```
+  
+### Traverse 
```

Comment:
The title should be `traverse_`, not to be confused with `traverse` in the `Traverse` typeclass. Maybe it's time to rename this, as `_` is a leftover convention from Haskell for when a function returns Unit. cc @raulraja